clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/StandardGceAttributeValidator.groovy

```diff
@@ -294,20 +294,26 @@ class StandardGceAttributeValidator {
 
   def validateInstanceType(String instanceType, String location, GoogleNamedAccountCredentials credentials) {
     validateNotEmpty(instanceType, "instanceType")
-    if (instanceType?.startsWith('custom')) {
+    if (instanceType?.contains('custom')) {
       validateCustomInstanceType(instanceType, location, credentials)
     }
   }
 
-  def customInstanceRegExp = /custom-\d{1,2}-\d{4,6}/
+  def customInstanceRegExp = /(.*?)-?custom-\d{1,2}-\d{4,6}/
 
   def validateCustomInstanceType(String instanceType, String location, GoogleNamedAccountCredentials credentials) {
     if (!(instanceType ==~ customInstanceRegExp)) {
-      errors.rejectValue("instanceType", "${context}.instanceType.invalid", "Custom instance string must match pattern /custom-\\d{1,2}-\\d{4,6}/.")
+      errors.rejectValue("instanceType", "${context}.instanceType.invalid", "Custom instance string must match pattern /(.*?)-?custom-\\d{1,2}-\\d{4,6}/.")
       return false
     }
 
-    def ( vCpuCount, memory ) = instanceType.split('-').tail().collect { it.toDouble() }
+    def vCpuCount, memory
+    if (instanceType.startsWith('custom-')) {
+      (vCpuCount, memory) = instanceType.split('-').tail().collect { it.toDouble() }
+    } else{
+      (vCpuCount, memory) = instanceType.split('-').tail().tail().collect { it.toDouble() }
+    }
+
```

Comment:
As long as we're using a regex here, we may as well actually use it and not switch to string splits halfway through, I suppose...

```
  def customInstanceRegExp = /(\w-)?custom-(\d{1,2})-(\d{4,6})/

  def validateCustomInstanceType(String instanceType, String location, GoogleNamedAccountCredentials credentials) {
    def customTypeMatcher = instanceType =~ customInstanceRegExp
    if (!customTypeMatcher.matches()) {
      errors.rejectValue("instanceType", "${context}.instanceType.invalid", "Custom instance string must match pattern $customInstanceRegExp.")
      return false
    }

    def vCpuCount = customTypeMatcher.group(2).toDouble()
    def memory = customTypeMatcher.group(3).toDouble()
```

Reply:
yes, it's better use regex groups, right regex `/(.*)-?custom-(\d{1,2})-(\d{4,6})/` because `(\w-)` return from `e2` only `2`. 

thanks!